api/src/main/java/org/apache/iceberg/PartitionSpec.java

```diff
@@ -625,7 +634,7 @@ PartitionSpec buildUnchecked() {
 
   static void checkCompatibility(PartitionSpec spec, Schema schema) {
     for (PartitionField field : spec.fields) {
-      Type sourceType = schema.findType(field.sourceId());
+      Type sourceType = schema.findType(field.sourceIds().get(0));
```

Comment:
We should be extending this logic as well to handle multi arg sources

Reply:
Yup. I'm doing a new update.

Reply:
I introduced new `canTransform(List)` and `getResultType(List)` to support multi-args in transforms.
I didn't remove the "old" methods, and I propose to deprecate it.
Thoughts ?